pkg/core/workflow_store.go

```diff
@@ -2,43 +2,104 @@ package core
 
 import (
 	"context"
+	"fmt"
 	"time"
 
 	"github.com/fuseml/fuseml-core/gen/workflow"
+	"github.com/fuseml/fuseml-core/pkg/domain"
 )
 
+// storableWorkflow holds the workflow and the codesets assigned to it
+type storableWorkflow struct {
+	// Workflow assigned to the codeset
+	workflow *workflow.Workflow
+	// Codesets assigned to the workflow
+	assignedCodesets []*domain.Codeset
```

Comment:
Great ! This is exactly what we talked about: model the assignments under the cover of an existing component, in this case the workflow store, because they're too small to warrant implementing a whole new component just to manage them.

Note for post-MVP: as mentioned in my other (long) comment, this will grow into a bigger concept that controls how workflows react more generally to new versions of their artifact inputs being published: it will be more than just codesets, and it will also need to cover some control information: policies allowing the user more control over how and when the events are triggering workflow runs.